core/src/main/java/org/apache/druid/java/util/common/guava/WrappingYielder.java

```diff
@@ -62,6 +62,9 @@ public OutType get()
       catch (Exception e) {
         t.addSuppressed(e);
       }
+      if (t instanceof RuntimeException) {
```

Comment:
what prompted these changes ?

Reply:
Ah, the 'withBaggage', https://github.com/apache/incubator-druid/pull/8981/files#diff-84792f9d3cefe47cbb471669dce2a276R145, caused all of the `RuntimeException` being thrown to be wrapped in an additional `RuntimeException` which didn't seem very useful. Otherwise, I would have needed to change [the expected cause of the expected exception in the tests](https://github.com/apache/incubator-druid/blob/master/core/src/test/java/org/apache/druid/java/util/common/guava/ParallelMergeCombiningSequenceTest.java#L452) to expect `RuntimeException` instead of `TimeoutException`.

Reply:
I see, in the surrounding code `Throwables.propagateIfPossible(t);` is used to get that behavior instead.

Reply:
Oops, that's a good point, I just threw these in here to fix my tests. Looking closer I will have a pass to clean this up a bit.